lib/core/timer.hpp

```diff
@@ -9,10 +10,17 @@ namespace hyped::core {
 class Timer {
  public:
   Timer(const ITimeSource &time);
-  Duration measure_execution_time(const std::function<void(void)> task);
+
+  Duration measureExecutionTime(const std::function<void(void)> task);
+
+  Duration measureElapsedTime(const TimePoint current_timepoint,
+                              const TimePoint previous_timepoint);
+
+  Float elapsedTimeInSeconds(const Duration time_elapsed);
```

Comment:
i feel like this wants a warning along the lines of "don't use when precision matters"

Reply:
Precision probably does matter though. Would a double be accurate enough?

Reply:
If you care about precision then you should return `Duration` as much as possible, that's nano-second precision

Reply:
i think this function is still fine for testing purposes and that sort of stuff; it just shouldn't be used in critical environments

Reply:
Yes but I need to multiply a duration value by a Float and we don't want ugly stuff like duration casts outside of core. What's the best way to do this?

Reply:
https://en.cppreference.com/w/cpp/chrono/duration/operator_arith4